index.bs

```diff
@@ -2334,6 +2309,121 @@ The [=default allowlist=] for this feature is <code>["self"]</code>.
 
 Note: If the document's origin is not allowed to use the <code>"xr-spatial-tracking"</code> feature policy any [=immersive sessions=] will be blocked, because all [=immersive sessions=] require some use of spatial tracking. [=Inline sessions=] will still be allowed, but restricted to only using the {{XRReferenceSpaceType/"viewer"}} {{XRReferenceSpace}}.
 
+
+Permissions API Integration {#permissions}
+---------
+
+The [[!permissions]] API provides a uniform way for websites to request permissions from users and query which permissions they have been granted. 
+
+The <dfn enum-value for="PermissionName">"xr"</dfn> [=powerful feature=]’s permission-related algorithms and types are defined as follows: 
+
+
+<dl>
+<dt>[=permission descriptor type=]</dt>
+<dd>
+
+<pre class="idl">
+dictionary XRPermissionDescriptor: PermissionDescriptor {
+  XRSession session;
+  sequence&lt;any&gt; requiredFeatures;
+  sequence&lt;any&gt; optionalFeatures;
+};
+</pre>
+
+{{PermissionDescriptor/name}} for {{XRPermissionDescriptor}} is {{PermissionName/"xr"}}.
+
+
+</dd>
+
+<dt>[=permission result type=]</dt>
+<dd>
+<pre class=idl>
+dictionary XRPermissionStatus: PermissionStatus {
+  sequence&lt;any&gt; granted;
+};
+</pre>
+
+</dd>
+
+
+<dt>[=permission request algorithm=]</dt>
+<dd>
+<div class=algorithm data-algorithm="xr-permission-request-algorithm">
+To <dfn lt="request the xr permission">request the "xr" permission</dfn> with an {{XRPermissionDescriptor}} |descriptor| and a {{XRPermissionStatus}} |status|, the UA MUST run the following steps:
+
+  1. Set |status|'s {{XRPermissionStatus/granted}} to an empty {{FrozenArray}}.
+  1. Let |result| be the result of [=resolve the requested features|resolving the requested features=] given |descriptor|'s {{XRPermissionDescriptor/requiredFeatures}}, {{XRPermissionDescriptor/optionalFeatures}}, and {{XRPermissionDescriptor/session}}.
+  1. If |result| is <code>null</code>, run the following steps:
+    1. Set |status|'s {{PermissionStatus/state}} to {{PermissionState/"denied"}}.
+    1. Abort these steps.
+  1. Let <code>(|consentRequired|, |consentOptional|, |granted|)</code> be the fields of |result|.
+  1. The user agent MAY at this point perform any necessary steps, like showing permissions prompts, to determine if there is a clear signal of [=user intent=] for any features in |requiredFeatures| and |optionalFeatures|.
+  1. For each |feature| in |consentRequired| perform the following steps:
+      1. The user agent MAY at this point perform any necessary steps, like showing permissions prompts, to determine if there is a clear signal of [=user intent=] for |feature|.
+      1. If a clear signal of [=user intent=] to enable |feature| has not been determined, set |status|'s {{PermissionStatus/state}} to {{PermissionState/"denied"}} and abort these steps.
```

Comment:
FWIW, the Permissions API has language for that. Maybe we can try to re-use it? #request-permission-to-use for example

Reply:
As mentioned above, this is not useful here, `#request-permission-to-use` takes a whole permission descriptor, we're requesting stuff for individual subfeatures. It would be nice if the permissions spec exposed a more scoped way to do prompts, but it doesn't, everything has to use a descriptor.

Reply:
Re-using the wording would be a first step.

Reply:
Fixed.